src/compile.ts

```diff
@@ -58,14 +58,20 @@ export function compile(
         if (isAppsyncResolver(node)) {
           return visitAppsyncResolver(node);
         } else if (isReflectFunction(node)) {
-          return toFunction("FunctionDecl", node.arguments[0]);
+          const func = node.arguments[0];
+          if (ts.isIdentifier(func)) {
+            throw new Error(
+              `Reflect does not support function references: ${func.text}`
+            );
+          }
+          return toFunction("FunctionDecl", func);
         }
         return ts.visitEachChild(node, visitor, ctx);
       }
 
       function isReflectFunction(node: ts.Node): node is ts.CallExpression & {
         arguments: [
-          ts.FunctionExpression | ts.ArrowFunction,
+          ts.FunctionExpression | ts.ArrowFunction | ts.Identifier,
```

Comment:
Why did we add support for this if it's an error?

Reply:
Because it is reality. Unless reflect fails, it can return all of the other types and the output should try to reflect reality, not intention.